components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APIAuthenticationHandler.java

```diff
@@ -445,8 +446,14 @@ public boolean handleRequest(MessageContext messageContext) {
                     log.error("API authentication failure due to "
                             + APISecurityConstants.API_AUTH_GENERAL_ERROR_MESSAGE, e);
                 } else {
+                    String warnDetails = getWarnDetails(messageContext);
                     // We do not need to log known authentication failures as errors since these are not product errors.
-                    log.warn("API authentication failure due to " + errorMessage);
+                    log.warn("API authentication failure due to " + errorMessage + warnDetails);
+                    // Sample Error String (JWT): APIAuthenticationHandler API authentication failure due to The access
```

Comment:
```suggestion
                    // Sample Warn String (JWT): APIAuthenticationHandler API authentication failure due to The access
```
move the comment before the warn log